src/__swaps__/screens/Swap/hooks/useSwapSettings.ts

```diff
@@ -14,6 +14,16 @@ export const useSwapSettings = ({ inputAsset }: { inputAsset: SharedValue<Extend
   const setSlippage = swapsStore(state => state.setSlippage);
   const setFlashbots = swapsStore(state => state.setFlashbots);
 
+  const checkIfFlashbotsIsEnabledOnSelectedInputAssetChain = (asset: ExtendedAnimatedAssetWithColors | null) => {
+    'worklet';
+
+    if (asset?.chainId === ChainId.mainnet) {
+      flashbots.value = true;
+    } else {
+      flashbots.value = false;
+    }
+  };
```

Comment:
important to note this doesn't effect persistence in the js store since it doesn't update that (which is intended).